compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -502,17 +502,27 @@ impl Printer<'tcx> for SymbolMangler<'tcx> {
         }
         let start = self.out.len();
 
-        match ct.ty.kind() {
-            ty::Uint(_) => {}
-            ty::Bool => {}
+        let mut neg = false;
+        let val = match ct.ty.kind() {
+            ty::Uint(_) | ty::Bool | ty::Char => {
+                ct.try_eval_bits(self.tcx, ty::ParamEnv::reveal_all(), ct.ty)
+            }
+            ty::Int(_) => {
+                neg = true;
+                ct.val
```

Comment:
I'm not sure whether this is the right approach.

Reply:
`neg = true` doesn't seem right when the value could be positive.

Reply:
I should stop writing code after midnight…